src/osd/OSD.h

```diff
@@ -1866,8 +1866,8 @@ class OSD : public Dispatcher,
   void got_full_map(epoch_t e);
 
   // -- failures --
-  std::map<int,utime_t> failure_queue;
-  std::map<int,std::pair<utime_t,entity_addrvec_t> > failure_pending;
+  std::map<int,std::pair<utime_t,int> > failure_queue; //{osd:<time,flags>}
+  std::map<int,boost::tuple<utime_t,entity_addrvec_t,int> > failure_pending; //{osd:<time,addr,flags>}
```

Comment:
Add failure_pending_t, failure_queue_t structs rather than using a tuple.  It'll be easier to read.